jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/AbstractConfiguration.java

```diff
@@ -16,131 +16,148 @@
+         *
+         * @param classes classname or package pattern
+         */
+        public Builder protect(String... classes)
+        {
+            _system.add(classes);
+            return this;
+        }
 
-    /**
-     * Expose classes to the web application by adding them
-     * as exclusions to the {@link WebAppConfiguration#getServerClasses()}
-     *
-     * @param classes classname or package pattern
-     */
-    protected void expose(String... classes)
-    {
-        for (String c : classes)
+        /**
+         * Hide classes from the web application by adding them
+         * to the {@link WebAppConfiguration#getServerClasses()}
+         *
+         * @param classes classname or package pattern
+         */
+        public Builder hide(String... classes)
         {
-            if (c.startsWith("-"))
-                throw new IllegalArgumentException();
-            _server.add("-" + c);
+            _server.add(classes);
+            return this;
         }
-    }
 
-    /**
-     * Protect classes from modification by the web application by adding them
-     * to the {@link WebAppConfiguration#getSystemClasses()} and
-     * expose them to the web application by adding them
-     * as exclusions to the {@link WebAppConfiguration#getServerClasses()}
-     *
-     * @param classes classname or package pattern
-     */
-    protected void protectAndExpose(String... classes)
-    {
-        for (String c : classes)
+        /**
+         * Expose classes to the web application by adding them
+         * as exclusions to the {@link WebAppConfiguration#getServerClasses()}
+         *
+         * @param classes classname or package pattern
+         */
+        public Builder expose(String... classes)
         {
-            if (c.startsWith("-"))
-                throw new IllegalArgumentException();
+            for (String c : classes)
+            {
+                if (c.startsWith("-"))
+                    throw new IllegalArgumentException();
+                _server.add("-" + c);
+            }
+            return this;
+        }
 
-            _system.add(c);
-            _server.add("-" + c);
+        /**
+         * Protect classes from modification by the web application by adding them
+         * to the {@link WebAppConfiguration#getSystemClasses()} and
+         * expose them to the web application by adding them
+         * as exclusions to the {@link WebAppConfiguration#getServerClasses()}
+         *
+         * @param classes classname or package pattern
+         */
+        public Builder protectAndExpose(String... classes)
+        {
+            for (String c : classes)
+            {
+                if (c.startsWith("-"))
+                    throw new IllegalArgumentException();
+
+                _system.add(c);
+                _server.add("-" + c);
+            }
+            return this;
         }
     }
 
+    protected AbstractConfiguration(Builder builder)
```

Comment:
This is a bit of a weird builder pattern. Normally you don't construct the object by passing it a builder, you use the builder to construct the object. Can't these fields just be `protected final` blank in `AbstractConfiguration`?

Reply:
It is a little different, but I don't think it can be miss-used, so I think it is OK.